torch/_inductor/kernel/flex_attention.py

```diff
@@ -573,9 +573,9 @@ def load_checked_2d(
     l_i = tl.where(l_i == 0.0, 1, l_i)
 
     acc = acc / l_i[:, None]
-    idx_zq = tl.program_id(1) // HQ
-    idx_hq = tl.program_id(1) % HQ
-    idx_m = offs_m[:, None]
+    idx_zq = (tl.program_id(1) // HQ).to(tl.int64)
```

Comment:
perf hit??

Reply:
actually did some testing and seems potentially fine.

Reply:
Yeah my first pass was I couldn't really measure but this was still exploratory, manman said that block_ptrs can have a meaningful perf impact still. I am going to swap to raw tl.load and see if thats the case

Reply:
just updated, will post perf runs